tests/rptest/tests/offset_for_leader_epoch_test.py

```diff
@@ -206,7 +206,10 @@ def have_all_offsets():
 
             wait_until(have_all_offsets, 30, 1)
 
-            return offsets_for_leader_epoch
+            return {
+                p.partition: p.epoch_end_offset
+                for p in offsets_for_leader_epoch
+            }
```

Comment:
nit: the function `get_offsets_for_leader_epoch` already looks overcomplicated, and we're adding to its complexity by  extracting these values twice: here and in the check. It would be nice to refactor it e.g. like this:
```
from rptest.util import wait_until_result
...
def get_offsets_for_leader_epoch(epoch):
    def get_all_offsets():
        kcl_res = kcl.offset_for_leader_epoch(topics=[topic.name], leader_epoch=epoch)
        extracted_res = [ofle.epoch_end_offset for ofle in kcl_res]
        return (False, None) if -1 in extracted_res else (True, extracted_res)
    return wait_until(get_all_offsets, 30, 1)
```

Reply:
cleaned it up.